### ISSUE TITLE
Windows system tray icon with connection status

### ISSUE THREAD

**Issue by rjrich** [symless/synergy-core#4800]
I relay on the Synergy icon on the menu bar in Mac OS X and on the panel in Linux Xfce to show me whether or not Synergy is connected. Apparently, the default in Synergy for Windows is not to show the Synergy icon in the Windows system tray. Could you please enable the option to show the Synergy icon in the Windows system tray?

Thank you.

**Comment by XinyuHou** [comment]
@rjrich 
It is a good idea. I did a brief research about how to implement it in Qt. It seems impossible.

Any suggestion?

**Comment by the-wes** [comment]
@XinyuHou - actually, the systray icon already does appear when the GUI is open. The problem is that the GUI process terminates when closing the window with the X. If I use Window -> Hide, it remains. So, I think we should have an option which the user can switch. This should make the behavior of the X button conditional. If option is set, it will do Hide. If it is not set, it will do Close.

This is a short term fix. In the longer term, there should be a systray icon which is not dependent on the GUI. Synergyc and Synergys already have this built-in, it's just turned off by default because the GUI issues --no-tray when spawning these processes. We could make this optional. This may even be easier and make more sense than my first suggestion.

Feel free to chat with me about this if you'd like to go into further detail.

**Comment by XinyuHou** [comment]
I think I misunderstood what @rjrich meant.

If I remember it correctly, press X button did hide the GUI. But with some reasons, we switched it to close. Can't remember exactly why?

**Comment by pkiff** [comment]
the-wes wrote:
> The problem is that the GUI process terminates when closing the window with the X. 
> If I use Window -> Hide, it remains. So, I think we should have an option which the user can switch.
Thanks for that simple note about how to force the tray icon to remain. I have long thought the tray icon should appear in the Windows version and didn't know why it didn't.

**Comment by badger200** [comment]
Yes this used to always work in Windows until recent years. Nowadays there seems like a total disconnect between the Synergy Client and the Synergy Service on Windows. 

For instance I currently have a functioning Synergy connection between my Windows XP workstation and Mac OS X 10.9. However I have no system tray for Synergy. The Mac status bar shows the Synergy icon with the green check.

If I open the Synergy Client on Windows XP, it acts like it's not connected. I can press Start. But this makes no sense because it is already running as a server/service and connected to the Mac OS X client. Furthermore, the system tray icon does not display the green check on Windows, even though it does on the Mac OS X client.

Ideally the software would be redesigned so this scenario is never possible. Synergy UI should always indicate the correct status of an existing connection.

**Comment by sunk818** [comment]
In Windows, Window > Hide and clicking on X should have the same behavior.

**Comment by szukai** [comment]
Not to mention if hidden double clicking the system tray should show the client ui window as well.

**Comment by szukai** [comment]
cry
